test/LanguageServer.Engine.Tests/TestProject.cs

```diff
@@ -0,0 +1,386 @@
+        {
+            if (testClassType == null)
+                throw new ArgumentNullException(nameof(testClassType));
+
+            if (relativePathSegments == null)
+                throw new ArgumentNullException(nameof(relativePathSegments));
+
+            return CreateProjectCollection(testClassType, logger: null, relativePathSegments);
+        }
+
+        /// <summary>
+        ///     Create an MSBuild <see cref="ProjectCollection"/> using the specified project directory.
+        /// </summary>
+        /// <param name="testClassType">
+        ///     The type of test-suite class used to determine the assembly containing the currently-running test (which can be used to find the test's deployment directory).
+        /// </param>
+        /// <param name="logger">
+        ///     An optional <see cref="ILogger"/> that will receives MSBuild logs.
+        /// </param>
+        /// <param name="relativePathSegments">
+        ///     The target project file's relative (to the test deployment directory) path segments.
+        /// </param>
+        /// <returns>
+        ///     The configured <see cref="ProjectCollection"/>.
+        /// </returns>
+        public static ProjectCollection CreateProjectCollection(Type testClassType, ILogger logger, params string[] relativePathSegments)
+        {
+            if (testClassType == null)
+                throw new ArgumentNullException(nameof(testClassType));
+
+            string projectDirectory = GetProjectDirectory(testClassType, relativePathSegments);
+
+            ProjectCollection projectCollection = MSBuildHelper.CreateProjectCollection(projectDirectory, logger: logger);
+
+            return projectCollection;
+        }
+
+        /// <summary>
+        ///     Load a test project into the project collection.
+        /// </summary>
+        /// <typeparam name="TTestClass">
+        ///     The test-suite class used to determine the assembly containing the currently-running test (which can be used to find the test's deployment directory).
+        /// </typeparam>
+        /// <param name="relativePathSegments">
+        ///     The project file's relative (to the test deployment directory) path segments.
+        /// </param>
+        /// <returns>
+        ///     The loaded project, as a <see cref="TestProject"/>.
+        /// </returns>
+        public static TestProject LoadTestProject<TTestClass>(this ProjectCollection projectCollection, params string[] relativePathSegments)
+            where TTestClass : class
+        {
+            if (projectCollection == null)
+                throw new ArgumentNullException(nameof(projectCollection));
+
+            if (relativePathSegments == null)
+                throw new ArgumentNullException(nameof(relativePathSegments));
+
+            Type testClassType = typeof(TTestClass);
+
+            return projectCollection.LoadTestProject(testClassType, relativePathSegments);
```

Comment:
```suggestion
            return projectCollection.LoadTestProject(typeof(TTestClass), relativePathSegments);
```
I don't see much value in extracting a single `typeof` operation into a local

Reply:
Mainly to be visually consistent with the other overloads and functions in this class, I guess. It makes sense to me, but I don’t feel too strongly about it so I’ll make a note to revisit it next time I add another test of this type.